crates/bevy_reflect/src/type_registry.rs

```diff
@@ -100,6 +104,28 @@ impl TypeRegistry {
             .insert(registration.type_id(), registration);
     }
 
+    /// Registers the type data `D` for type `T`.
+    ///
+    /// Most of the time [`TypeRegistry::register`] can be used instead to register a type you derived [`Reflect`] for.
+    /// However, in cases where you want to add a piece of type data that was not included in the list of `#[reflect(...)]` type data in the derive,
+    /// or where the type is generic and cannot register e.g. `ReflectSerialize` unconditionally without knowing the specific type parameters,
+    /// this method can be used to insert additional type data.
+    ///
+    /// # Example
```

Comment:
Why return `Option`? Should this return a `Result` that errors if the type is not yet registered (and gives a helper error message as to what went wrong)?

Reply:
I would agree here. The fully explicit registering approach is better in this case imo.
when I first started working with Reflection I definitely relied on "Magic", and I wouldn't be surprised if others did too especially in this case. Panicking early with a helpful message once is much better than silent failures later, and directs the user to correct long-term usage of this API.